From f4a021d554083ae89d5c0bbe005f7b99848a3b37 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm.code@gmail.com>
Date: Mon, 23 Apr 2018 22:56:08 +0200
Subject: [PATCH 09/22] build: Use minimal compression when zipping targetfiles

This saves about 10 seconds in packaging time at the expense of about
8% compression loss:

Using -L 5 (default):
  targetfiles size 1392mb
  compression time 21.8s

Using -L 1:
  targetfiles size 1508mb
  compression time 11.4s

This is a good tradeoff, as the targetfiles zip is just an intermediate
package which is used to generate the final OTA zip.

Change-Id: I7479c723bcd4dbf73690a2f3cc31b10dff49062f
---
 core/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index 4ad81a261..dfa7b8e44 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -2742,7 +2742,7 @@ endif
 	@# help early validation of the .zip file while uploading it.
 	$(hide) find $(zip_root)/META | sort >$@.list
 	$(hide) find $(zip_root) -path $(zip_root)/META -prune -o -print | sort >>$@.list
-	$(hide) $(SOONG_ZIP) -d -o $@ -C $(zip_root) -l $@.list
+	$(hide) $(SOONG_ZIP) -L 1 -d -o $@ -C $(zip_root) -l $@.list
 
 .PHONY: target-files-package
 target-files-package: $(BUILT_TARGET_FILES_PACKAGE)
-- 
2.17.1

